[DOC] Add deprecation for metrics summary (#4193)
grafana/tempo: docs/sources/tempo/api_docs/metrics-summary.md

# Metrics summary API

{{< admonition type="warning" >}}

> Metrics summary API is deprecated in Grafana Cloud and Grafana 11.3 and later.

Reviewer: since these are OSS tempo docs should we mention a tempo version (2.7?) instead of a grafana version and cloud?

Author: Good question. I removed the mention of Aggregate by because we don't discuss the UI here. What if we changed this to something like: 
```suggestion
Metrics summary API is deprecated in Tempo 2.7. Features powered by metrics summary API, like the Aggregate by table, are also deprecated in Grafana Cloud and Grafana 11.3 and later.
```

Reviewer: i like that 👍

> The metrics summary API is deprecated in Tempo 2.7. Features powered by the metrics summary API, like the Aggregate by table, are also deprecated in Grafana Cloud and Grafana 11.3 and later.

Reviewer 2: ```suggestion
The metrics summary API is deprecated as of Tempo 2.7. Features powered by the metrics summary API, like the Aggregate by table, are also deprecated in Grafana Cloud and Grafana 11.3 and later.
```

Reviewer 2: Not sure if this is more readable? Feel free to reject.

Author: I like that better. Thank you for the suggestion.

It will be removed in a future release.
{{% /admonition %}}

This document explains how to use the metrics summary API in Tempo.
This API returns RED metrics (span count, erroring span count, and latency information) for `kind=server` spans sent to Tempo in the last hour, grouped by a user-specified attribute.

{{< youtube id="g97CjKOZqT4" >}}

## Activate metrics summary

> To enable the experimental metrics summary API, you must turn on the local blocks processor in the metrics generator.

Reviewer 2: ```suggestion
To enable the (deprecated) metrics summary API, you must turn on the local blocks processor in the metrics generator.
```

      processors: [..., 'local-blocks']
```

> In Grafana and Grafana Cloudl, the Metrics summary API is an [experimental feature]({{< relref "/docs/release-life-cycle" >}}) that is disabled by default.

Author: ```suggestion
In Grafana and Grafana Cloudl, the Metrics summary API is an [experimental feature](https://grafana.com/docs/release-life-cycle/) that is disabled by default.
```

> In Grafana and Grafana Cloudl, the Metrics summary API is an [experimental feature](https://grafana.com/docs/release-life-cycle/) that is disabled by default.

Reviewer 2: ```suggestion
In Grafana and Grafana Cloud, the Metrics summary API is an [experimental feature](https://grafana.com/docs/release-life-cycle/) that is disabled by default.
```

Reviewer 2: Now that its deprecated, we should no longer call it experimental but rather 'deprecated'.

Reviewer 2: Makes me think we should have a "deprecated" entry on the 'release lifecycle' page: https://grafana.com/docs/release-life-cycle/

But that is a story for another day. :)

Author: We should. We do for the docs: https://grafana.com/docs/writers-toolkit/write/deprecate-remove/
